theme/theme.inc

```diff
@@ -32,6 +34,12 @@ function islandora_internet_archive_bookreader_preprocess_islandora_internet_arc
           'token' => $token,
         ),
       )),
+      'backupUri' => url("islandora/object/{$page_pid}/datastream/{$backup_dsid}/view", array(
+        'absolute' => TRUE,
+        'query' => array(
+          'token' => $backup_token,
```

Comment:
We can have a single callback to check size and return new url with a token if needed.